### PR TITLE
support enum type

### PR DESCRIPTION
**Pull request by YANGGMM**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #6613 

## What this PR does / why we need it:
support enum type

### CODE REVIEW

File: pkg/util/export/etl/tae.go [hunk 1]
```diff
@@ -400,7 +400,7 @@ func GetVectorArrayLen(ctx context.Context, vec *vector.Vector) (int, error) {
 		cols := vector.MustFixedCol[types.Datetime](vec)
 		return len(cols), nil
 	default:
-		return 0, moerr.NewInternalError(ctx, "the value type %d is not support now", *vec.GetType())
```
Comment (xzxiong) [inline pkg/util/export/etl/tae.go:403]: lgtm

**Review by xzxiong — APPROVED** [review]

**Review by daviszhen — APPROVED** [review]

**Review by XuPeng-SH — CHANGES_REQUESTED** [review]
This PR has a compatibility issue.

**Review by reusee — APPROVED** [review]

**Review by ouyuanning — APPROVED** [review]

**Review by qingxinhome — APPROVED** [review]

**Review by nnsgmsone — APPROVED** [review]

**Review by heni02 — APPROVED** [review]

**Review by triump2020 — APPROVED** [review]

**Review by m-schen — APPROVED** [review]
后续如果要支持enum类型的函数的话可能需要在函数框架改点代码。

**Review by aunjgr — APPROVED** [review]

File: pkg/sql/plan/function/type_check.go [hunk 2]
```diff
@@ -1482,6 +1483,7 @@ func initFixed3() {
 				{toType: types.T_varbinary, preferLevel: 2},
 				{toType: types.T_blob, preferLevel: 2},
 				{toType: types.T_text, preferLevel: 2},
+				{toType: types.T_enum, preferLevel: 2},
```
[the remark this answers is not in the document]
Comment (YANGGMM) [reply pkg/sql/plan/function/type_check.go:1486]: OK, thanks

File: pkg/vectorize/enum/enum.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/matrixorigin/matrixone/pkg/common/moerr"
+)
+
+// ParseEnum return item index with item name or value.
+func ParseEnum(elems []string, name string) (uint16, error) {
```
[the remark this answers is not in the document]
Comment (YANGGMM) [reply pkg/vectorize/enum/enum.go]: OK, thanks

[the remark this answers is not in the document]
Comment (YANGGMM) [reply pkg/vectorize/enum/enum.go]: 之前不是跟函数有关的调用方法都放在那个目录下面

**Comment by YANGGMM** [comment]
> 后续如果要支持enum类型的函数的话可能需要在函数框架改点代码。

好的，如果有需求的话会继续修改，谢谢

**Review by LeftHandCold — APPROVED** [review]

**Review by XuPeng-SH — APPROVED** [review]

**Pull request merged**